lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1411,7 +1411,35 @@ def generate_students_certificates(
     start_time = time()
     enrolled_students = CourseEnrollment.objects.users_enrolled_in(course_id)
 
-    students = task_input.get('students', None)
+    which_students = task_input.get('students', None)
+    if which_students == 'all_whitelisted':
+        # Generate Certificates for all white listed students.
+        students = User.objects.filter(
+            certificatewhitelist__course_id=course_id,
+            certificatewhitelist__whitelist=True
+        ).values_list('id', flat=True)
+
+    elif which_students == 'whitelisted_not_generated':
+        # All Whitelisted students
+        all_whitelisted_students = User.objects.filter(
+            certificatewhitelist__course_id=course_id,
+            certificatewhitelist__whitelist=True
+        )
+
+        # Whitelisted students which got certificates already.
+        certificate_generated_students = GeneratedCertificate.objects.filter(  # pylint: disable=no-member
+            user__in=all_whitelisted_students,
+            course_id=course_id,
+        )
+
+        all_whitelisted_students_ids = set(all_whitelisted_students.values_list('id', flat=True))
+        certificate_generated_students_ids = set(certificate_generated_students.values_list('user_id', flat=True))
+        certificate_not_generated_ids = list(all_whitelisted_students_ids - certificate_generated_students_ids)
+
+        # Whitelisted students which does not have certificates yet.
+        students = all_whitelisted_students.filter(id__in=certificate_not_generated_ids).values_list('id', flat=True)
+    else:
+        students = which_students
 
     if students is not None:
         enrolled_students = enrolled_students.filter(id__in=students)
```

Comment:
I think this logic can be improved.

1) this variable should not be named `enrolled_students` — it should be `students_to_generate_certs_for` or something like that.
2) It doesn't make sense to do multiple queries for the `all_whitelisted` and `whitelisted_not_generated` alternatives:
`all_whitelisted`:
```python
students_to_generate_certs_for.filter(
    courseenrollment__course_id=course_id,
    courseenrollment__is_active=True
)
```
For `whitelisted_not_generated`, you can make more complex queries by leveraging django's [`Q` objects](https://docs.djangoproject.com/en/1.9/topics/db/queries/#complex-lookups-with-q-objects).

Finally, for the case of a specific student, you can do something like `students_to_generate_certs_for.filter(id=specific_student_id)`